{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0ba71418-e16b-4102-97a2-15bc045eda1c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "library(\"RColorBrewer\")\n",
    "library(\"data.table\")\n",
    "library(\"readxl\")\n",
    "library(\"writexl\")\n",
    "library(\"stringr\")\n",
    "library(\"tidyr\")\n",
    "library(\"dplyr\")\n",
    "library(\"tidyverse\")\n",
    "library(\"rlang\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "38cc1826-a45c-447b-90ce-5e2ace6dd113",
   "metadata": {},
   "outputs": [],
   "source": [
    "orig_svginfo = readRDS(\"../gitrepos/kinome-tree/Data/kintree.RDS\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b28ebf23-9449-4fbb-9120-c909ed3ee38e",
   "metadata": {},
   "outputs": [],
   "source": [
    "dt<-read.csv(\"./tempfiles/output-kdata.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "64ec42d1-940d-44ec-8159-c73673ed7fac",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 22 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>X</th><th scope=col>SYMBOL</th><th scope=col>Just_Kinase</th><th scope=col>Result</th><th scope=col>HGNC_SYMBOL</th><th scope=col>CRO_Kinase</th><th scope=col>bin_25</th><th scope=col>bin_10</th><th scope=col>bin_25_range</th><th scope=col>bin_10_range</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 1</td><td>CDK3   </td><td>CDK3/CYCE1</td><td>103.85340</td><td>CDK3   </td><td>CDK3/CYCE1</td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td> 2</td><td>CDK6   </td><td>CDK6/CYCD3</td><td>102.91460</td><td>CDK6   </td><td>CDK6/CYCD3</td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td> 3</td><td>CDK2   </td><td>CDK2/CYCA2</td><td>102.10290</td><td>CDK2   </td><td>CDK2/CYCA2</td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td> 4</td><td>CDK4   </td><td>CDK4/CYCD3</td><td>100.73582</td><td>CDK4   </td><td>CDK4/CYCD3</td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td> 5</td><td>PKD1   </td><td>PKD1      </td><td>100.59809</td><td>PKD1   </td><td>PKD1      </td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td> 6</td><td>DYRK2  </td><td>DYRK2     </td><td>100.04111</td><td>DYRK2  </td><td>DYRK2     </td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td> 7</td><td>MAPK7  </td><td>ERK5      </td><td> 98.69070</td><td>MAPK7  </td><td>ERK5      </td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td> 8</td><td>DYRK3  </td><td>DYRK3     </td><td> 97.99615</td><td>DYRK3  </td><td>DYRK3     </td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td> 9</td><td>PKD3   </td><td>PKD3      </td><td> 95.28954</td><td>PKD3   </td><td>PKD3      </td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td>10</td><td>CDK5   </td><td>CDK5/P25  </td><td> 95.11848</td><td>CDK5   </td><td>CDK5/P25  </td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td>11</td><td>CDK1   </td><td>CDK1/CYCB1</td><td> 94.86562</td><td>CDK1   </td><td>CDK1/CYCB1</td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td>12</td><td>ROCK2  </td><td>ROCK2     </td><td> 94.70350</td><td>ROCK2  </td><td>ROCK2     </td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td>13</td><td>STK24  </td><td>MST3      </td><td> 86.44842</td><td>STK24  </td><td>MST3      </td><td> 90</td><td> 90</td><td>76-90 </td><td>81-90 </td></tr>\n",
       "\t<tr><td>14</td><td>RPS6KA3</td><td>RSK2      </td><td> 72.07244</td><td>RPS6KA3</td><td>RSK2      </td><td> 75</td><td> 80</td><td>51-75 </td><td>71-80 </td></tr>\n",
       "\t<tr><td>15</td><td>STK26  </td><td>MST4      </td><td> 71.48551</td><td>STK26  </td><td>MST4      </td><td> 75</td><td> 80</td><td>51-75 </td><td>71-80 </td></tr>\n",
       "\t<tr><td>16</td><td>MYLK2  </td><td>SKMLCK    </td><td> 70.69339</td><td>MYLK2  </td><td>SKMLCK    </td><td> 75</td><td> 80</td><td>51-75 </td><td>71-80 </td></tr>\n",
       "\t<tr><td>17</td><td>PRKCD  </td><td>PKCD      </td><td> 68.21224</td><td>PRKCD  </td><td>PKCD      </td><td> 75</td><td> 70</td><td>51-75 </td><td>61-70 </td></tr>\n",
       "\t<tr><td>18</td><td>MKNK2  </td><td>MNK2      </td><td> 67.15418</td><td>MKNK2  </td><td>MNK2      </td><td> 75</td><td> 70</td><td>51-75 </td><td>61-70 </td></tr>\n",
       "\t<tr><td>19</td><td>MKNK1  </td><td>MNK1      </td><td> 63.62229</td><td>MKNK1  </td><td>MNK1      </td><td> 75</td><td> 70</td><td>51-75 </td><td>61-70 </td></tr>\n",
       "\t<tr><td>20</td><td>CDK9   </td><td>CDK9/CYCT1</td><td> 60.45794</td><td>CDK9   </td><td>CDK9/CYCT1</td><td> 75</td><td> 70</td><td>51-75 </td><td>61-70 </td></tr>\n",
       "\t<tr><td>21</td><td>CLK3   </td><td>CLK3      </td><td> 52.39962</td><td>CLK3   </td><td>CLK3      </td><td> 75</td><td> 60</td><td>51-75 </td><td>51-60 </td></tr>\n",
       "\t<tr><td>22</td><td>GSK3B  </td><td>GSK3B     </td><td> 52.24657</td><td>GSK3B  </td><td>GSK3B     </td><td> 75</td><td> 60</td><td>51-75 </td><td>51-60 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 22 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " X & SYMBOL & Just\\_Kinase & Result & HGNC\\_SYMBOL & CRO\\_Kinase & bin\\_25 & bin\\_10 & bin\\_25\\_range & bin\\_10\\_range\\\\\n",
       " <int> & <chr> & <chr> & <dbl> & <chr> & <chr> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t  1 & CDK3    & CDK3/CYCE1 & 103.85340 & CDK3    & CDK3/CYCE1 & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t  2 & CDK6    & CDK6/CYCD3 & 102.91460 & CDK6    & CDK6/CYCD3 & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t  3 & CDK2    & CDK2/CYCA2 & 102.10290 & CDK2    & CDK2/CYCA2 & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t  4 & CDK4    & CDK4/CYCD3 & 100.73582 & CDK4    & CDK4/CYCD3 & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t  5 & PKD1    & PKD1       & 100.59809 & PKD1    & PKD1       & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t  6 & DYRK2   & DYRK2      & 100.04111 & DYRK2   & DYRK2      & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t  7 & MAPK7   & ERK5       &  98.69070 & MAPK7   & ERK5       & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t  8 & DYRK3   & DYRK3      &  97.99615 & DYRK3   & DYRK3      & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t  9 & PKD3    & PKD3       &  95.28954 & PKD3    & PKD3       & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t 10 & CDK5    & CDK5/P25   &  95.11848 & CDK5    & CDK5/P25   & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t 11 & CDK1    & CDK1/CYCB1 &  94.86562 & CDK1    & CDK1/CYCB1 & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t 12 & ROCK2   & ROCK2      &  94.70350 & ROCK2   & ROCK2      & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t 13 & STK24   & MST3       &  86.44842 & STK24   & MST3       &  90 &  90 & 76-90  & 81-90 \\\\\n",
       "\t 14 & RPS6KA3 & RSK2       &  72.07244 & RPS6KA3 & RSK2       &  75 &  80 & 51-75  & 71-80 \\\\\n",
       "\t 15 & STK26   & MST4       &  71.48551 & STK26   & MST4       &  75 &  80 & 51-75  & 71-80 \\\\\n",
       "\t 16 & MYLK2   & SKMLCK     &  70.69339 & MYLK2   & SKMLCK     &  75 &  80 & 51-75  & 71-80 \\\\\n",
       "\t 17 & PRKCD   & PKCD       &  68.21224 & PRKCD   & PKCD       &  75 &  70 & 51-75  & 61-70 \\\\\n",
       "\t 18 & MKNK2   & MNK2       &  67.15418 & MKNK2   & MNK2       &  75 &  70 & 51-75  & 61-70 \\\\\n",
       "\t 19 & MKNK1   & MNK1       &  63.62229 & MKNK1   & MNK1       &  75 &  70 & 51-75  & 61-70 \\\\\n",
       "\t 20 & CDK9    & CDK9/CYCT1 &  60.45794 & CDK9    & CDK9/CYCT1 &  75 &  70 & 51-75  & 61-70 \\\\\n",
       "\t 21 & CLK3    & CLK3       &  52.39962 & CLK3    & CLK3       &  75 &  60 & 51-75  & 51-60 \\\\\n",
       "\t 22 & GSK3B   & GSK3B      &  52.24657 & GSK3B   & GSK3B      &  75 &  60 & 51-75  & 51-60 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 22 × 10\n",
       "\n",
       "| X &lt;int&gt; | SYMBOL &lt;chr&gt; | Just_Kinase &lt;chr&gt; | Result &lt;dbl&gt; | HGNC_SYMBOL &lt;chr&gt; | CRO_Kinase &lt;chr&gt; | bin_25 &lt;int&gt; | bin_10 &lt;int&gt; | bin_25_range &lt;chr&gt; | bin_10_range &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "|  1 | CDK3    | CDK3/CYCE1 | 103.85340 | CDK3    | CDK3/CYCE1 | 100 | 100 | 90-100 | 91-100 |\n",
       "|  2 | CDK6    | CDK6/CYCD3 | 102.91460 | CDK6    | CDK6/CYCD3 | 100 | 100 | 90-100 | 91-100 |\n",
       "|  3 | CDK2    | CDK2/CYCA2 | 102.10290 | CDK2    | CDK2/CYCA2 | 100 | 100 | 90-100 | 91-100 |\n",
       "|  4 | CDK4    | CDK4/CYCD3 | 100.73582 | CDK4    | CDK4/CYCD3 | 100 | 100 | 90-100 | 91-100 |\n",
       "|  5 | PKD1    | PKD1       | 100.59809 | PKD1    | PKD1       | 100 | 100 | 90-100 | 91-100 |\n",
       "|  6 | DYRK2   | DYRK2      | 100.04111 | DYRK2   | DYRK2      | 100 | 100 | 90-100 | 91-100 |\n",
       "|  7 | MAPK7   | ERK5       |  98.69070 | MAPK7   | ERK5       | 100 | 100 | 90-100 | 91-100 |\n",
       "|  8 | DYRK3   | DYRK3      |  97.99615 | DYRK3   | DYRK3      | 100 | 100 | 90-100 | 91-100 |\n",
       "|  9 | PKD3    | PKD3       |  95.28954 | PKD3    | PKD3       | 100 | 100 | 90-100 | 91-100 |\n",
       "| 10 | CDK5    | CDK5/P25   |  95.11848 | CDK5    | CDK5/P25   | 100 | 100 | 90-100 | 91-100 |\n",
       "| 11 | CDK1    | CDK1/CYCB1 |  94.86562 | CDK1    | CDK1/CYCB1 | 100 | 100 | 90-100 | 91-100 |\n",
       "| 12 | ROCK2   | ROCK2      |  94.70350 | ROCK2   | ROCK2      | 100 | 100 | 90-100 | 91-100 |\n",
       "| 13 | STK24   | MST3       |  86.44842 | STK24   | MST3       |  90 |  90 | 76-90  | 81-90  |\n",
       "| 14 | RPS6KA3 | RSK2       |  72.07244 | RPS6KA3 | RSK2       |  75 |  80 | 51-75  | 71-80  |\n",
       "| 15 | STK26   | MST4       |  71.48551 | STK26   | MST4       |  75 |  80 | 51-75  | 71-80  |\n",
       "| 16 | MYLK2   | SKMLCK     |  70.69339 | MYLK2   | SKMLCK     |  75 |  80 | 51-75  | 71-80  |\n",
       "| 17 | PRKCD   | PKCD       |  68.21224 | PRKCD   | PKCD       |  75 |  70 | 51-75  | 61-70  |\n",
       "| 18 | MKNK2   | MNK2       |  67.15418 | MKNK2   | MNK2       |  75 |  70 | 51-75  | 61-70  |\n",
       "| 19 | MKNK1   | MNK1       |  63.62229 | MKNK1   | MNK1       |  75 |  70 | 51-75  | 61-70  |\n",
       "| 20 | CDK9    | CDK9/CYCT1 |  60.45794 | CDK9    | CDK9/CYCT1 |  75 |  70 | 51-75  | 61-70  |\n",
       "| 21 | CLK3    | CLK3       |  52.39962 | CLK3    | CLK3       |  75 |  60 | 51-75  | 51-60  |\n",
       "| 22 | GSK3B   | GSK3B      |  52.24657 | GSK3B   | GSK3B      |  75 |  60 | 51-75  | 51-60  |\n",
       "\n"
      ],
      "text/plain": [
       "   X  SYMBOL  Just_Kinase Result    HGNC_SYMBOL CRO_Kinase bin_25 bin_10\n",
       "1   1 CDK3    CDK3/CYCE1  103.85340 CDK3        CDK3/CYCE1 100    100   \n",
       "2   2 CDK6    CDK6/CYCD3  102.91460 CDK6        CDK6/CYCD3 100    100   \n",
       "3   3 CDK2    CDK2/CYCA2  102.10290 CDK2        CDK2/CYCA2 100    100   \n",
       "4   4 CDK4    CDK4/CYCD3  100.73582 CDK4        CDK4/CYCD3 100    100   \n",
       "5   5 PKD1    PKD1        100.59809 PKD1        PKD1       100    100   \n",
       "6   6 DYRK2   DYRK2       100.04111 DYRK2       DYRK2      100    100   \n",
       "7   7 MAPK7   ERK5         98.69070 MAPK7       ERK5       100    100   \n",
       "8   8 DYRK3   DYRK3        97.99615 DYRK3       DYRK3      100    100   \n",
       "9   9 PKD3    PKD3         95.28954 PKD3        PKD3       100    100   \n",
       "10 10 CDK5    CDK5/P25     95.11848 CDK5        CDK5/P25   100    100   \n",
       "11 11 CDK1    CDK1/CYCB1   94.86562 CDK1        CDK1/CYCB1 100    100   \n",
       "12 12 ROCK2   ROCK2        94.70350 ROCK2       ROCK2      100    100   \n",
       "13 13 STK24   MST3         86.44842 STK24       MST3        90     90   \n",
       "14 14 RPS6KA3 RSK2         72.07244 RPS6KA3     RSK2        75     80   \n",
       "15 15 STK26   MST4         71.48551 STK26       MST4        75     80   \n",
       "16 16 MYLK2   SKMLCK       70.69339 MYLK2       SKMLCK      75     80   \n",
       "17 17 PRKCD   PKCD         68.21224 PRKCD       PKCD        75     70   \n",
       "18 18 MKNK2   MNK2         67.15418 MKNK2       MNK2        75     70   \n",
       "19 19 MKNK1   MNK1         63.62229 MKNK1       MNK1        75     70   \n",
       "20 20 CDK9    CDK9/CYCT1   60.45794 CDK9        CDK9/CYCT1  75     70   \n",
       "21 21 CLK3    CLK3         52.39962 CLK3        CLK3        75     60   \n",
       "22 22 GSK3B   GSK3B        52.24657 GSK3B       GSK3B       75     60   \n",
       "   bin_25_range bin_10_range\n",
       "1  90-100       91-100      \n",
       "2  90-100       91-100      \n",
       "3  90-100       91-100      \n",
       "4  90-100       91-100      \n",
       "5  90-100       91-100      \n",
       "6  90-100       91-100      \n",
       "7  90-100       91-100      \n",
       "8  90-100       91-100      \n",
       "9  90-100       91-100      \n",
       "10 90-100       91-100      \n",
       "11 90-100       91-100      \n",
       "12 90-100       91-100      \n",
       "13 76-90        81-90       \n",
       "14 51-75        71-80       \n",
       "15 51-75        71-80       \n",
       "16 51-75        71-80       \n",
       "17 51-75        61-70       \n",
       "18 51-75        61-70       \n",
       "19 51-75        61-70       \n",
       "20 51-75        61-70       \n",
       "21 51-75        51-60       \n",
       "22 51-75        51-60       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "103f9ab5-69fb-4fc9-beb8-38d9264a44f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'CDK3'</li><li>'CDK6'</li><li>'CDK2'</li><li>'CDK4'</li><li>'PKD1'</li><li>'DYRK2'</li><li>'MAPK7'</li><li>'DYRK3'</li><li>'PKD3'</li><li>'CDK5'</li><li>'CDK1'</li><li>'ROCK2'</li><li>'STK24'</li><li>'RPS6KA3'</li><li>'STK26'</li><li>'MYLK2'</li><li>'PRKCD'</li><li>'MKNK2'</li><li>'MKNK1'</li><li>'CDK9'</li><li>'CLK3'</li><li>'GSK3B'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CDK3'\n",
       "\\item 'CDK6'\n",
       "\\item 'CDK2'\n",
       "\\item 'CDK4'\n",
       "\\item 'PKD1'\n",
       "\\item 'DYRK2'\n",
       "\\item 'MAPK7'\n",
       "\\item 'DYRK3'\n",
       "\\item 'PKD3'\n",
       "\\item 'CDK5'\n",
       "\\item 'CDK1'\n",
       "\\item 'ROCK2'\n",
       "\\item 'STK24'\n",
       "\\item 'RPS6KA3'\n",
       "\\item 'STK26'\n",
       "\\item 'MYLK2'\n",
       "\\item 'PRKCD'\n",
       "\\item 'MKNK2'\n",
       "\\item 'MKNK1'\n",
       "\\item 'CDK9'\n",
       "\\item 'CLK3'\n",
       "\\item 'GSK3B'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CDK3'\n",
       "2. 'CDK6'\n",
       "3. 'CDK2'\n",
       "4. 'CDK4'\n",
       "5. 'PKD1'\n",
       "6. 'DYRK2'\n",
       "7. 'MAPK7'\n",
       "8. 'DYRK3'\n",
       "9. 'PKD3'\n",
       "10. 'CDK5'\n",
       "11. 'CDK1'\n",
       "12. 'ROCK2'\n",
       "13. 'STK24'\n",
       "14. 'RPS6KA3'\n",
       "15. 'STK26'\n",
       "16. 'MYLK2'\n",
       "17. 'PRKCD'\n",
       "18. 'MKNK2'\n",
       "19. 'MKNK1'\n",
       "20. 'CDK9'\n",
       "21. 'CLK3'\n",
       "22. 'GSK3B'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"CDK3\"    \"CDK6\"    \"CDK2\"    \"CDK4\"    \"PKD1\"    \"DYRK2\"   \"MAPK7\"  \n",
       " [8] \"DYRK3\"   \"PKD3\"    \"CDK5\"    \"CDK1\"    \"ROCK2\"   \"STK24\"   \"RPS6KA3\"\n",
       "[15] \"STK26\"   \"MYLK2\"   \"PRKCD\"   \"MKNK2\"   \"MKNK1\"   \"CDK9\"    \"CLK3\"   \n",
       "[22] \"GSK3B\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dt$HGNC_SYMBOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "0da89652-3a38-4d15-b01a-5a6238add67d",
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- orig_svginfo$dataframe %>% \n",
    "        select(id.coral, id.HGNC) %>%\n",
    "        merge(dt, by.x = \"id.HGNC\", by.y = \"HGNC_SYMBOL\") %>%\n",
    "        mutate(KINASE = id.coral, HGNC_SYMBOL = id.HGNC) %>%\n",
    "        arrange(HGNC_SYMBOL)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "32ae6d72-7472-44b2-9472-6f09797ab663",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "21"
      ],
      "text/latex": [
       "21"
      ],
      "text/markdown": [
       "21"
      ],
      "text/plain": [
       "[1] 21"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(df)[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "1c262086-3ba5-4e3c-b83e-ece3423e64cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "22"
      ],
      "text/latex": [
       "22"
      ],
      "text/markdown": [
       "22"
      ],
      "text/plain": [
       "[1] 22"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(dt)[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "23a7b755-f456-488c-a2d8-e6f1dde4752b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "dt <- dt %>% rename(id.HGNC= HGNC_SYMBOL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "4a4c2c2e-92bc-4942-9114-b7b952791faa",
   "metadata": {},
   "outputs": [],
   "source": [
    "missing <- dt %>%\n",
    "        anti_join(orig_svginfo$dataframe, by = \"id.HGNC\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "aa351f39-5eed-41ea-8cd8-590ae24afc39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>X</th><th scope=col>SYMBOL</th><th scope=col>Just_Kinase</th><th scope=col>Result</th><th scope=col>id.HGNC</th><th scope=col>CRO_Kinase</th><th scope=col>bin_25</th><th scope=col>bin_10</th><th scope=col>bin_25_range</th><th scope=col>bin_10_range</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>5</td><td>PKD1</td><td>PKD1</td><td>100.59809</td><td>PKD1</td><td>PKD1</td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "\t<tr><td>9</td><td>PKD3</td><td>PKD3</td><td> 95.28954</td><td>PKD3</td><td>PKD3</td><td>100</td><td>100</td><td>90-100</td><td>91-100</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " X & SYMBOL & Just\\_Kinase & Result & id.HGNC & CRO\\_Kinase & bin\\_25 & bin\\_10 & bin\\_25\\_range & bin\\_10\\_range\\\\\n",
       " <int> & <chr> & <chr> & <dbl> & <chr> & <chr> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 5 & PKD1 & PKD1 & 100.59809 & PKD1 & PKD1 & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\t 9 & PKD3 & PKD3 &  95.28954 & PKD3 & PKD3 & 100 & 100 & 90-100 & 91-100\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 10\n",
       "\n",
       "| X &lt;int&gt; | SYMBOL &lt;chr&gt; | Just_Kinase &lt;chr&gt; | Result &lt;dbl&gt; | id.HGNC &lt;chr&gt; | CRO_Kinase &lt;chr&gt; | bin_25 &lt;int&gt; | bin_10 &lt;int&gt; | bin_25_range &lt;chr&gt; | bin_10_range &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 5 | PKD1 | PKD1 | 100.59809 | PKD1 | PKD1 | 100 | 100 | 90-100 | 91-100 |\n",
       "| 9 | PKD3 | PKD3 |  95.28954 | PKD3 | PKD3 | 100 | 100 | 90-100 | 91-100 |\n",
       "\n"
      ],
      "text/plain": [
       "  X SYMBOL Just_Kinase Result    id.HGNC CRO_Kinase bin_25 bin_10 bin_25_range\n",
       "1 5 PKD1   PKD1        100.59809 PKD1    PKD1       100    100    90-100      \n",
       "2 9 PKD3   PKD3         95.28954 PKD3    PKD3       100    100    90-100      \n",
       "  bin_10_range\n",
       "1 91-100      \n",
       "2 91-100      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "missing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "08ec1875-c05c-45ee-920d-4b4793296f53",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>id.coral</th><th scope=col>id.HGNC</th><th scope=col>X</th><th scope=col>SYMBOL</th><th scope=col>Just_Kinase</th><th scope=col>Result</th><th scope=col>CRO_Kinase</th><th scope=col>bin_25</th><th scope=col>bin_10</th><th scope=col>bin_25_range</th><th scope=col>bin_10_range</th><th scope=col>KINASE</th><th scope=col>HGNC_SYMBOL</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>PKD1</td><td>PRKD1</td><td>5</td><td>PKD1</td><td>PKD1</td><td>100.59809</td><td>PKD1</td><td>100</td><td>100</td><td>90-100</td><td>91-100</td><td>PKD1</td><td>PRKD1</td></tr>\n",
       "\t<tr><td>PKD3</td><td>PRKD3</td><td>9</td><td>PKD3</td><td>PKD3</td><td> 95.28954</td><td>PKD3</td><td>100</td><td>100</td><td>90-100</td><td>91-100</td><td>PKD3</td><td>PRKD3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " id.coral & id.HGNC & X & SYMBOL & Just\\_Kinase & Result & CRO\\_Kinase & bin\\_25 & bin\\_10 & bin\\_25\\_range & bin\\_10\\_range & KINASE & HGNC\\_SYMBOL\\\\\n",
       " <chr> & <chr> & <int> & <chr> & <chr> & <dbl> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t PKD1 & PRKD1 & 5 & PKD1 & PKD1 & 100.59809 & PKD1 & 100 & 100 & 90-100 & 91-100 & PKD1 & PRKD1\\\\\n",
       "\t PKD3 & PRKD3 & 9 & PKD3 & PKD3 &  95.28954 & PKD3 & 100 & 100 & 90-100 & 91-100 & PKD3 & PRKD3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 13\n",
       "\n",
       "| id.coral &lt;chr&gt; | id.HGNC &lt;chr&gt; | X &lt;int&gt; | SYMBOL &lt;chr&gt; | Just_Kinase &lt;chr&gt; | Result &lt;dbl&gt; | CRO_Kinase &lt;chr&gt; | bin_25 &lt;int&gt; | bin_10 &lt;int&gt; | bin_25_range &lt;chr&gt; | bin_10_range &lt;chr&gt; | KINASE &lt;chr&gt; | HGNC_SYMBOL &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| PKD1 | PRKD1 | 5 | PKD1 | PKD1 | 100.59809 | PKD1 | 100 | 100 | 90-100 | 91-100 | PKD1 | PRKD1 |\n",
       "| PKD3 | PRKD3 | 9 | PKD3 | PKD3 |  95.28954 | PKD3 | 100 | 100 | 90-100 | 91-100 | PKD3 | PRKD3 |\n",
       "\n"
      ],
      "text/plain": [
       "  id.coral id.HGNC X SYMBOL Just_Kinase Result    CRO_Kinase bin_25 bin_10\n",
       "1 PKD1     PRKD1   5 PKD1   PKD1        100.59809 PKD1       100    100   \n",
       "2 PKD3     PRKD3   9 PKD3   PKD3         95.28954 PKD3       100    100   \n",
       "  bin_25_range bin_10_range KINASE HGNC_SYMBOL\n",
       "1 90-100       91-100       PKD1   PRKD1      \n",
       "2 90-100       91-100       PKD3   PRKD3      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "orig_svginfo$dataframe %>%\n",
    "    select(id.coral, id.HGNC) %>%\n",
    "    merge(missing, by.x = \"id.coral\", by.y = \"id.HGNC\") %>%\n",
    "    mutate(KINASE = id.coral, HGNC_SYMBOL = id.HGNC) %>%\n",
    "    arrange(HGNC_SYMBOL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "1bb213ef-362e-4cfd-bb82-1faf005506fc",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1</li><li>2</li><li>3</li><li>4</li><li>12</li><li>13</li><li>14</li><li>15</li><li>20</li><li>21</li><li>22</li><li>25</li><li>26</li><li>27</li><li>28</li><li>31</li><li>32</li><li>33</li><li>34</li><li>37</li><li>38</li><li>39</li><li>43</li><li>44</li><li>45</li><li>46</li><li>47</li><li>48</li><li>49</li><li>52</li><li>53</li><li>54</li><li>55</li><li>56</li><li>57</li><li>58</li><li>59</li><li>60</li><li>61</li><li>62</li><li>63</li><li>66</li><li>67</li><li>68</li><li>69</li><li>71</li><li>82</li><li>83</li><li>84</li><li>85</li><li>86</li><li>91</li><li>92</li><li>93</li><li>94</li><li>95</li><li>96</li><li>97</li><li>98</li><li>99</li><li>100</li><li>101</li><li>102</li><li>103</li><li>104</li><li>105</li><li>106</li><li>107</li><li>108</li><li>110</li><li>111</li><li>112</li><li>113</li><li>114</li><li>115</li><li>116</li><li>117</li><li>118</li><li>119</li><li>120</li><li>121</li><li>126</li><li>127</li><li>133</li><li>134</li><li>135</li><li>136</li><li>137</li><li>138</li><li>139</li><li>140</li><li>141</li><li>142</li><li>143</li><li>144</li><li>145</li><li>146</li><li>147</li><li>148</li><li>149</li><li>150</li><li>157</li><li>158</li><li>159</li><li>160</li><li>161</li><li>162</li><li>163</li><li>164</li><li>165</li><li>166</li><li>167</li><li>168</li><li>170</li><li>171</li><li>172</li><li>173</li><li>176</li><li>181</li><li>182</li><li>183</li><li>184</li><li>192</li><li>193</li><li>195</li><li>196</li><li>197</li><li>198</li><li>199</li><li>201</li><li>202</li><li>203</li><li>204</li><li>205</li><li>206</li><li>207</li><li>208</li><li>209</li><li>210</li><li>211</li><li>212</li><li>213</li><li>214</li><li>215</li><li>219</li><li>222</li><li>223</li><li>225</li><li>226</li><li>227</li><li>228</li><li>232</li><li>233</li><li>234</li><li>236</li><li>237</li><li>238</li><li>239</li><li>241</li><li>242</li><li>243</li><li>244</li><li>245</li><li>246</li><li>247</li><li>248</li><li>249</li><li>250</li><li>251</li><li>252</li><li>253</li><li>254</li><li>255</li><li>256</li><li>264</li><li>265</li><li>266</li><li>267</li><li>268</li><li>270</li><li>271</li><li>278</li><li>281</li><li>282</li><li>283</li><li>290</li><li>295</li><li>298</li><li>299</li><li>300</li><li>301</li><li>302</li><li>303</li><li>304</li><li>305</li><li>306</li><li>307</li><li>308</li><li>309</li><li>310</li><li>311</li><li>312</li><li>315</li><li>316</li><li>320</li><li>321</li><li>336</li><li>340</li><li>341</li><li>342</li><li>343</li><li>344</li><li>345</li><li>349</li><li>350</li><li>351</li><li>352</li><li>353</li><li>354</li><li>355</li><li>357</li><li>358</li><li>359</li><li>369</li><li>370</li><li>371</li><li>372</li><li>373</li><li>374</li><li>375</li><li>376</li><li>377</li><li>378</li><li>379</li><li>380</li><li>381</li><li>382</li><li>383</li><li>385</li><li>386</li><li>387</li><li>388</li><li>391</li><li>392</li><li>397</li><li>398</li><li>399</li><li>400</li><li>403</li><li>404</li><li>405</li><li>406</li><li>415</li><li>416</li><li>417</li><li>418</li><li>419</li><li>420</li><li>421</li><li>422</li><li>423</li><li>424</li><li>425</li><li>426</li><li>427</li><li>428</li><li>429</li><li>430</li><li>431</li><li>432</li><li>433</li><li>434</li><li>435</li><li>436</li><li>437</li><li>438</li><li>439</li><li>441</li><li>443</li><li>445</li><li>447</li><li>448</li><li>449</li><li>450</li><li>451</li><li>452</li><li>453</li><li>454</li><li>458</li><li>459</li><li>464</li><li>465</li><li>466</li><li>471</li><li>472</li><li>473</li><li>474</li><li>475</li><li>476</li><li>477</li><li>479</li><li>480</li><li>481</li><li>482</li><li>483</li><li>484</li><li>485</li><li>486</li><li>487</li><li>489</li><li>490</li><li>491</li><li>492</li><li>493</li><li>494</li><li>495</li><li>496</li><li>497</li><li>498</li><li>499</li><li>500</li><li>501</li><li>502</li><li>503</li><li>504</li><li>505</li><li>506</li><li>507</li><li>508</li><li>509</li><li>510</li><li>511</li><li>512</li><li>513</li><li>514</li><li>515</li><li>516</li><li>517</li><li>518</li><li>521</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 20\n",
       "\\item 21\n",
       "\\item 22\n",
       "\\item 25\n",
       "\\item 26\n",
       "\\item 27\n",
       "\\item 28\n",
       "\\item 31\n",
       "\\item 32\n",
       "\\item 33\n",
       "\\item 34\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 39\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 52\n",
       "\\item 53\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 56\n",
       "\\item 57\n",
       "\\item 58\n",
       "\\item 59\n",
       "\\item 60\n",
       "\\item 61\n",
       "\\item 62\n",
       "\\item 63\n",
       "\\item 66\n",
       "\\item 67\n",
       "\\item 68\n",
       "\\item 69\n",
       "\\item 71\n",
       "\\item 82\n",
       "\\item 83\n",
       "\\item 84\n",
       "\\item 85\n",
       "\\item 86\n",
       "\\item 91\n",
       "\\item 92\n",
       "\\item 93\n",
       "\\item 94\n",
       "\\item 95\n",
       "\\item 96\n",
       "\\item 97\n",
       "\\item 98\n",
       "\\item 99\n",
       "\\item 100\n",
       "\\item 101\n",
       "\\item 102\n",
       "\\item 103\n",
       "\\item 104\n",
       "\\item 105\n",
       "\\item 106\n",
       "\\item 107\n",
       "\\item 108\n",
       "\\item 110\n",
       "\\item 111\n",
       "\\item 112\n",
       "\\item 113\n",
       "\\item 114\n",
       "\\item 115\n",
       "\\item 116\n",
       "\\item 117\n",
       "\\item 118\n",
       "\\item 119\n",
       "\\item 120\n",
       "\\item 121\n",
       "\\item 126\n",
       "\\item 127\n",
       "\\item 133\n",
       "\\item 134\n",
       "\\item 135\n",
       "\\item 136\n",
       "\\item 137\n",
       "\\item 138\n",
       "\\item 139\n",
       "\\item 140\n",
       "\\item 141\n",
       "\\item 142\n",
       "\\item 143\n",
       "\\item 144\n",
       "\\item 145\n",
       "\\item 146\n",
       "\\item 147\n",
       "\\item 148\n",
       "\\item 149\n",
       "\\item 150\n",
       "\\item 157\n",
       "\\item 158\n",
       "\\item 159\n",
       "\\item 160\n",
       "\\item 161\n",
       "\\item 162\n",
       "\\item 163\n",
       "\\item 164\n",
       "\\item 165\n",
       "\\item 166\n",
       "\\item 167\n",
       "\\item 168\n",
       "\\item 170\n",
       "\\item 171\n",
       "\\item 172\n",
       "\\item 173\n",
       "\\item 176\n",
       "\\item 181\n",
       "\\item 182\n",
       "\\item 183\n",
       "\\item 184\n",
       "\\item 192\n",
       "\\item 193\n",
       "\\item 195\n",
       "\\item 196\n",
       "\\item 197\n",
       "\\item 198\n",
       "\\item 199\n",
       "\\item 201\n",
       "\\item 202\n",
       "\\item 203\n",
       "\\item 204\n",
       "\\item 205\n",
       "\\item 206\n",
       "\\item 207\n",
       "\\item 208\n",
       "\\item 209\n",
       "\\item 210\n",
       "\\item 211\n",
       "\\item 212\n",
       "\\item 213\n",
       "\\item 214\n",
       "\\item 215\n",
       "\\item 219\n",
       "\\item 222\n",
       "\\item 223\n",
       "\\item 225\n",
       "\\item 226\n",
       "\\item 227\n",
       "\\item 228\n",
       "\\item 232\n",
       "\\item 233\n",
       "\\item 234\n",
       "\\item 236\n",
       "\\item 237\n",
       "\\item 238\n",
       "\\item 239\n",
       "\\item 241\n",
       "\\item 242\n",
       "\\item 243\n",
       "\\item 244\n",
       "\\item 245\n",
       "\\item 246\n",
       "\\item 247\n",
       "\\item 248\n",
       "\\item 249\n",
       "\\item 250\n",
       "\\item 251\n",
       "\\item 252\n",
       "\\item 253\n",
       "\\item 254\n",
       "\\item 255\n",
       "\\item 256\n",
       "\\item 264\n",
       "\\item 265\n",
       "\\item 266\n",
       "\\item 267\n",
       "\\item 268\n",
       "\\item 270\n",
       "\\item 271\n",
       "\\item 278\n",
       "\\item 281\n",
       "\\item 282\n",
       "\\item 283\n",
       "\\item 290\n",
       "\\item 295\n",
       "\\item 298\n",
       "\\item 299\n",
       "\\item 300\n",
       "\\item 301\n",
       "\\item 302\n",
       "\\item 303\n",
       "\\item 304\n",
       "\\item 305\n",
       "\\item 306\n",
       "\\item 307\n",
       "\\item 308\n",
       "\\item 309\n",
       "\\item 310\n",
       "\\item 311\n",
       "\\item 312\n",
       "\\item 315\n",
       "\\item 316\n",
       "\\item 320\n",
       "\\item 321\n",
       "\\item 336\n",
       "\\item 340\n",
       "\\item 341\n",
       "\\item 342\n",
       "\\item 343\n",
       "\\item 344\n",
       "\\item 345\n",
       "\\item 349\n",
       "\\item 350\n",
       "\\item 351\n",
       "\\item 352\n",
       "\\item 353\n",
       "\\item 354\n",
       "\\item 355\n",
       "\\item 357\n",
       "\\item 358\n",
       "\\item 359\n",
       "\\item 369\n",
       "\\item 370\n",
       "\\item 371\n",
       "\\item 372\n",
       "\\item 373\n",
       "\\item 374\n",
       "\\item 375\n",
       "\\item 376\n",
       "\\item 377\n",
       "\\item 378\n",
       "\\item 379\n",
       "\\item 380\n",
       "\\item 381\n",
       "\\item 382\n",
       "\\item 383\n",
       "\\item 385\n",
       "\\item 386\n",
       "\\item 387\n",
       "\\item 388\n",
       "\\item 391\n",
       "\\item 392\n",
       "\\item 397\n",
       "\\item 398\n",
       "\\item 399\n",
       "\\item 400\n",
       "\\item 403\n",
       "\\item 404\n",
       "\\item 405\n",
       "\\item 406\n",
       "\\item 415\n",
       "\\item 416\n",
       "\\item 417\n",
       "\\item 418\n",
       "\\item 419\n",
       "\\item 420\n",
       "\\item 421\n",
       "\\item 422\n",
       "\\item 423\n",
       "\\item 424\n",
       "\\item 425\n",
       "\\item 426\n",
       "\\item 427\n",
       "\\item 428\n",
       "\\item 429\n",
       "\\item 430\n",
       "\\item 431\n",
       "\\item 432\n",
       "\\item 433\n",
       "\\item 434\n",
       "\\item 435\n",
       "\\item 436\n",
       "\\item 437\n",
       "\\item 438\n",
       "\\item 439\n",
       "\\item 441\n",
       "\\item 443\n",
       "\\item 445\n",
       "\\item 447\n",
       "\\item 448\n",
       "\\item 449\n",
       "\\item 450\n",
       "\\item 451\n",
       "\\item 452\n",
       "\\item 453\n",
       "\\item 454\n",
       "\\item 458\n",
       "\\item 459\n",
       "\\item 464\n",
       "\\item 465\n",
       "\\item 466\n",
       "\\item 471\n",
       "\\item 472\n",
       "\\item 473\n",
       "\\item 474\n",
       "\\item 475\n",
       "\\item 476\n",
       "\\item 477\n",
       "\\item 479\n",
       "\\item 480\n",
       "\\item 481\n",
       "\\item 482\n",
       "\\item 483\n",
       "\\item 484\n",
       "\\item 485\n",
       "\\item 486\n",
       "\\item 487\n",
       "\\item 489\n",
       "\\item 490\n",
       "\\item 491\n",
       "\\item 492\n",
       "\\item 493\n",
       "\\item 494\n",
       "\\item 495\n",
       "\\item 496\n",
       "\\item 497\n",
       "\\item 498\n",
       "\\item 499\n",
       "\\item 500\n",
       "\\item 501\n",
       "\\item 502\n",
       "\\item 503\n",
       "\\item 504\n",
       "\\item 505\n",
       "\\item 506\n",
       "\\item 507\n",
       "\\item 508\n",
       "\\item 509\n",
       "\\item 510\n",
       "\\item 511\n",
       "\\item 512\n",
       "\\item 513\n",
       "\\item 514\n",
       "\\item 515\n",
       "\\item 516\n",
       "\\item 517\n",
       "\\item 518\n",
       "\\item 521\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1\n",
       "2. 2\n",
       "3. 3\n",
       "4. 4\n",
       "5. 12\n",
       "6. 13\n",
       "7. 14\n",
       "8. 15\n",
       "9. 20\n",
       "10. 21\n",
       "11. 22\n",
       "12. 25\n",
       "13. 26\n",
       "14. 27\n",
       "15. 28\n",
       "16. 31\n",
       "17. 32\n",
       "18. 33\n",
       "19. 34\n",
       "20. 37\n",
       "21. 38\n",
       "22. 39\n",
       "23. 43\n",
       "24. 44\n",
       "25. 45\n",
       "26. 46\n",
       "27. 47\n",
       "28. 48\n",
       "29. 49\n",
       "30. 52\n",
       "31. 53\n",
       "32. 54\n",
       "33. 55\n",
       "34. 56\n",
       "35. 57\n",
       "36. 58\n",
       "37. 59\n",
       "38. 60\n",
       "39. 61\n",
       "40. 62\n",
       "41. 63\n",
       "42. 66\n",
       "43. 67\n",
       "44. 68\n",
       "45. 69\n",
       "46. 71\n",
       "47. 82\n",
       "48. 83\n",
       "49. 84\n",
       "50. 85\n",
       "51. 86\n",
       "52. 91\n",
       "53. 92\n",
       "54. 93\n",
       "55. 94\n",
       "56. 95\n",
       "57. 96\n",
       "58. 97\n",
       "59. 98\n",
       "60. 99\n",
       "61. 100\n",
       "62. 101\n",
       "63. 102\n",
       "64. 103\n",
       "65. 104\n",
       "66. 105\n",
       "67. 106\n",
       "68. 107\n",
       "69. 108\n",
       "70. 110\n",
       "71. 111\n",
       "72. 112\n",
       "73. 113\n",
       "74. 114\n",
       "75. 115\n",
       "76. 116\n",
       "77. 117\n",
       "78. 118\n",
       "79. 119\n",
       "80. 120\n",
       "81. 121\n",
       "82. 126\n",
       "83. 127\n",
       "84. 133\n",
       "85. 134\n",
       "86. 135\n",
       "87. 136\n",
       "88. 137\n",
       "89. 138\n",
       "90. 139\n",
       "91. 140\n",
       "92. 141\n",
       "93. 142\n",
       "94. 143\n",
       "95. 144\n",
       "96. 145\n",
       "97. 146\n",
       "98. 147\n",
       "99. 148\n",
       "100. 149\n",
       "101. 150\n",
       "102. 157\n",
       "103. 158\n",
       "104. 159\n",
       "105. 160\n",
       "106. 161\n",
       "107. 162\n",
       "108. 163\n",
       "109. 164\n",
       "110. 165\n",
       "111. 166\n",
       "112. 167\n",
       "113. 168\n",
       "114. 170\n",
       "115. 171\n",
       "116. 172\n",
       "117. 173\n",
       "118. 176\n",
       "119. 181\n",
       "120. 182\n",
       "121. 183\n",
       "122. 184\n",
       "123. 192\n",
       "124. 193\n",
       "125. 195\n",
       "126. 196\n",
       "127. 197\n",
       "128. 198\n",
       "129. 199\n",
       "130. 201\n",
       "131. 202\n",
       "132. 203\n",
       "133. 204\n",
       "134. 205\n",
       "135. 206\n",
       "136. 207\n",
       "137. 208\n",
       "138. 209\n",
       "139. 210\n",
       "140. 211\n",
       "141. 212\n",
       "142. 213\n",
       "143. 214\n",
       "144. 215\n",
       "145. 219\n",
       "146. 222\n",
       "147. 223\n",
       "148. 225\n",
       "149. 226\n",
       "150. 227\n",
       "151. 228\n",
       "152. 232\n",
       "153. 233\n",
       "154. 234\n",
       "155. 236\n",
       "156. 237\n",
       "157. 238\n",
       "158. 239\n",
       "159. 241\n",
       "160. 242\n",
       "161. 243\n",
       "162. 244\n",
       "163. 245\n",
       "164. 246\n",
       "165. 247\n",
       "166. 248\n",
       "167. 249\n",
       "168. 250\n",
       "169. 251\n",
       "170. 252\n",
       "171. 253\n",
       "172. 254\n",
       "173. 255\n",
       "174. 256\n",
       "175. 264\n",
       "176. 265\n",
       "177. 266\n",
       "178. 267\n",
       "179. 268\n",
       "180. 270\n",
       "181. 271\n",
       "182. 278\n",
       "183. 281\n",
       "184. 282\n",
       "185. 283\n",
       "186. 290\n",
       "187. 295\n",
       "188. 298\n",
       "189. 299\n",
       "190. 300\n",
       "191. 301\n",
       "192. 302\n",
       "193. 303\n",
       "194. 304\n",
       "195. 305\n",
       "196. 306\n",
       "197. 307\n",
       "198. 308\n",
       "199. 309\n",
       "200. 310\n",
       "201. 311\n",
       "202. 312\n",
       "203. 315\n",
       "204. 316\n",
       "205. 320\n",
       "206. 321\n",
       "207. 336\n",
       "208. 340\n",
       "209. 341\n",
       "210. 342\n",
       "211. 343\n",
       "212. 344\n",
       "213. 345\n",
       "214. 349\n",
       "215. 350\n",
       "216. 351\n",
       "217. 352\n",
       "218. 353\n",
       "219. 354\n",
       "220. 355\n",
       "221. 357\n",
       "222. 358\n",
       "223. 359\n",
       "224. 369\n",
       "225. 370\n",
       "226. 371\n",
       "227. 372\n",
       "228. 373\n",
       "229. 374\n",
       "230. 375\n",
       "231. 376\n",
       "232. 377\n",
       "233. 378\n",
       "234. 379\n",
       "235. 380\n",
       "236. 381\n",
       "237. 382\n",
       "238. 383\n",
       "239. 385\n",
       "240. 386\n",
       "241. 387\n",
       "242. 388\n",
       "243. 391\n",
       "244. 392\n",
       "245. 397\n",
       "246. 398\n",
       "247. 399\n",
       "248. 400\n",
       "249. 403\n",
       "250. 404\n",
       "251. 405\n",
       "252. 406\n",
       "253. 415\n",
       "254. 416\n",
       "255. 417\n",
       "256. 418\n",
       "257. 419\n",
       "258. 420\n",
       "259. 421\n",
       "260. 422\n",
       "261. 423\n",
       "262. 424\n",
       "263. 425\n",
       "264. 426\n",
       "265. 427\n",
       "266. 428\n",
       "267. 429\n",
       "268. 430\n",
       "269. 431\n",
       "270. 432\n",
       "271. 433\n",
       "272. 434\n",
       "273. 435\n",
       "274. 436\n",
       "275. 437\n",
       "276. 438\n",
       "277. 439\n",
       "278. 441\n",
       "279. 443\n",
       "280. 445\n",
       "281. 447\n",
       "282. 448\n",
       "283. 449\n",
       "284. 450\n",
       "285. 451\n",
       "286. 452\n",
       "287. 453\n",
       "288. 454\n",
       "289. 458\n",
       "290. 459\n",
       "291. 464\n",
       "292. 465\n",
       "293. 466\n",
       "294. 471\n",
       "295. 472\n",
       "296. 473\n",
       "297. 474\n",
       "298. 475\n",
       "299. 476\n",
       "300. 477\n",
       "301. 479\n",
       "302. 480\n",
       "303. 481\n",
       "304. 482\n",
       "305. 483\n",
       "306. 484\n",
       "307. 485\n",
       "308. 486\n",
       "309. 487\n",
       "310. 489\n",
       "311. 490\n",
       "312. 491\n",
       "313. 492\n",
       "314. 493\n",
       "315. 494\n",
       "316. 495\n",
       "317. 496\n",
       "318. 497\n",
       "319. 498\n",
       "320. 499\n",
       "321. 500\n",
       "322. 501\n",
       "323. 502\n",
       "324. 503\n",
       "325. 504\n",
       "326. 505\n",
       "327. 506\n",
       "328. 507\n",
       "329. 508\n",
       "330. 509\n",
       "331. 510\n",
       "332. 511\n",
       "333. 512\n",
       "334. 513\n",
       "335. 514\n",
       "336. 515\n",
       "337. 516\n",
       "338. 517\n",
       "339. 518\n",
       "340. 521\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1]   1   2   3   4  12  13  14  15  20  21  22  25  26  27  28  31  32  33\n",
       " [19]  34  37  38  39  43  44  45  46  47  48  49  52  53  54  55  56  57  58\n",
       " [37]  59  60  61  62  63  66  67  68  69  71  82  83  84  85  86  91  92  93\n",
       " [55]  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 110 111 112\n",
       " [73] 113 114 115 116 117 118 119 120 121 126 127 133 134 135 136 137 138 139\n",
       " [91] 140 141 142 143 144 145 146 147 148 149 150 157 158 159 160 161 162 163\n",
       "[109] 164 165 166 167 168 170 171 172 173 176 181 182 183 184 192 193 195 196\n",
       "[127] 197 198 199 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
       "[145] 219 222 223 225 226 227 228 232 233 234 236 237 238 239 241 242 243 244\n",
       "[163] 245 246 247 248 249 250 251 252 253 254 255 256 264 265 266 267 268 270\n",
       "[181] 271 278 281 282 283 290 295 298 299 300 301 302 303 304 305 306 307 308\n",
       "[199] 309 310 311 312 315 316 320 321 336 340 341 342 343 344 345 349 350 351\n",
       "[217] 352 353 354 355 357 358 359 369 370 371 372 373 374 375 376 377 378 379\n",
       "[235] 380 381 382 383 385 386 387 388 391 392 397 398 399 400 403 404 405 406\n",
       "[253] 415 416 417 418 419 420 421 422 423 424 425 426 427 428 429 430 431 432\n",
       "[271] 433 434 435 436 437 438 439 441 443 445 447 448 449 450 451 452 453 454\n",
       "[289] 458 459 464 465 466 471 472 473 474 475 476 477 479 480 481 482 483 484\n",
       "[307] 485 486 487 489 490 491 492 493 494 495 496 497 498 499 500 501 502 503\n",
       "[325] 504 505 506 507 508 509 510 511 512 513 514 515 516 517 518 521"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "NAs = which(is.na(orig_svginfo$dataframe$kinase.subfamily))\n",
    "NAs # the NAs index within the kinaste.subfamily column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "09e7bce9-1094-4022-947a-33d3b8ffe028",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#head(orig_svginfo$dataframe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "4b57e20e-9b77-47f3-ba15-e2b1d585d95d",
   "metadata": {},
   "outputs": [],
   "source": [
    "orig_svginfo$dataframe$kinase.subfamily[NAs] = \"\" # set to empty string and not NA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "604f5b55-c861-4fc9-b6c2-cc88fcb78983",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove NAs from HGNCs\n",
    "NAs = which(is.na(orig_svginfo$dataframe$id.HGNC))\n",
    "orig_svginfo$dataframe$id.HGNC[NAs] = \"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "690ac783-f3b4-460e-b78e-e5cfb82ac744",
   "metadata": {},
   "outputs": [],
   "source": [
    "HGNC <- data.table(read_excel(\"../gitrepos//kinome-tree/data-input/HGNC-protein-coding-genes.xlsx\"))\n",
    "#setDT(HGNC)\n",
    "\n",
    "manual_map <- data.table(read_excel(\"../gitrepos//kinome-tree/data-input/manual-map.xlsx\"))\n",
    "#setDT(manual_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "b9d7a748-05eb-4d79-8f28-a505ff41b337",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'HGNC_ID'</li><li>'SYMBOL'</li><li>'NAME'</li><li>'ALIAS_SYMBOL'</li><li>'ALIAS_NAME'</li><li>'PREV_SYMBOL'</li><li>'PREV_NAME'</li><li>'GENE_FAMILY'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'HGNC\\_ID'\n",
       "\\item 'SYMBOL'\n",
       "\\item 'NAME'\n",
       "\\item 'ALIAS\\_SYMBOL'\n",
       "\\item 'ALIAS\\_NAME'\n",
       "\\item 'PREV\\_SYMBOL'\n",
       "\\item 'PREV\\_NAME'\n",
       "\\item 'GENE\\_FAMILY'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'HGNC_ID'\n",
       "2. 'SYMBOL'\n",
       "3. 'NAME'\n",
       "4. 'ALIAS_SYMBOL'\n",
       "5. 'ALIAS_NAME'\n",
       "6. 'PREV_SYMBOL'\n",
       "7. 'PREV_NAME'\n",
       "8. 'GENE_FAMILY'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"HGNC_ID\"      \"SYMBOL\"       \"NAME\"         \"ALIAS_SYMBOL\" \"ALIAS_NAME\"  \n",
       "[6] \"PREV_SYMBOL\"  \"PREV_NAME\"    \"GENE_FAMILY\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(HGNC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "ea83b5ea-7c9a-4c29-99bc-7c5fa7631f9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# kinasedata <- data.table(readxl::read_xlsx(\"../../Downloads/CBS-220074_FT007843_kinome_data.xlsx\"))\n",
    "kinasedata <- read.csv(\"../../Downloads/FT007615.csv\")\n",
    "#kinasedata2 <- data.table(readxl::read_xlsx(\"../../Downloads/test4.xlsx\"))\n",
    "# kinasedata <- read.csv(file = \"../data_files/ft007426_df.csv\", stringsAsFactors = F)# %>%\n",
    "                # select(KINASE, PCT_INHIBITION_AVG) %>% rename(Kinase = KINASE, Result = PCT_INHIBITION_AVG)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "efb41c10-6e58-49ad-8ced-66549b06b464",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>BATCH_ID</th><th scope=col>BATCH_NUM</th><th scope=col>COMPD_ID</th><th scope=col>CONC</th><th scope=col>EXPERIMENT_ID</th><th scope=col>KINASE</th><th scope=col>PCT_INHIBITION_AVG</th><th scope=col>TECHNOLOGY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET                   </td><td>99.330</td><td>Hot Spot</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET_L1195F            </td><td>99.020</td><td>Hot Spot</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET_D1228Y            </td><td>98.955</td><td>Hot Spot</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET_M1250I            </td><td>98.760</td><td>Hot Spot</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET-TFG               </td><td>98.510</td><td>Hot Spot</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET_D1228H            </td><td>98.395</td><td>Hot Spot</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET_D1228N            </td><td>97.520</td><td>Hot Spot</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET-KIF5B (Kex24Mex14)</td><td>97.020</td><td>Hot Spot</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET_D1228V            </td><td>96.990</td><td>Hot Spot</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>FT007615-05</td><td>5</td><td>FT007615</td><td>100</td><td>200904</td><td>MET_Y1230S            </td><td>96.985</td><td>Hot Spot</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & BATCH\\_ID & BATCH\\_NUM & COMPD\\_ID & CONC & EXPERIMENT\\_ID & KINASE & PCT\\_INHIBITION\\_AVG & TECHNOLOGY\\\\\n",
       "  & <chr> & <int> & <chr> & <int> & <int> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET                    & 99.330 & Hot Spot\\\\\n",
       "\t2 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET\\_L1195F             & 99.020 & Hot Spot\\\\\n",
       "\t3 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET\\_D1228Y             & 98.955 & Hot Spot\\\\\n",
       "\t4 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET\\_M1250I             & 98.760 & Hot Spot\\\\\n",
       "\t5 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET-TFG                & 98.510 & Hot Spot\\\\\n",
       "\t6 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET\\_D1228H             & 98.395 & Hot Spot\\\\\n",
       "\t7 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET\\_D1228N             & 97.520 & Hot Spot\\\\\n",
       "\t8 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET-KIF5B (Kex24Mex14) & 97.020 & Hot Spot\\\\\n",
       "\t9 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET\\_D1228V             & 96.990 & Hot Spot\\\\\n",
       "\t10 & FT007615-05 & 5 & FT007615 & 100 & 200904 & MET\\_Y1230S             & 96.985 & Hot Spot\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 8\n",
       "\n",
       "| <!--/--> | BATCH_ID &lt;chr&gt; | BATCH_NUM &lt;int&gt; | COMPD_ID &lt;chr&gt; | CONC &lt;int&gt; | EXPERIMENT_ID &lt;int&gt; | KINASE &lt;chr&gt; | PCT_INHIBITION_AVG &lt;dbl&gt; | TECHNOLOGY &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET                    | 99.330 | Hot Spot |\n",
       "| 2 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET_L1195F             | 99.020 | Hot Spot |\n",
       "| 3 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET_D1228Y             | 98.955 | Hot Spot |\n",
       "| 4 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET_M1250I             | 98.760 | Hot Spot |\n",
       "| 5 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET-TFG                | 98.510 | Hot Spot |\n",
       "| 6 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET_D1228H             | 98.395 | Hot Spot |\n",
       "| 7 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET_D1228N             | 97.520 | Hot Spot |\n",
       "| 8 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET-KIF5B (Kex24Mex14) | 97.020 | Hot Spot |\n",
       "| 9 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET_D1228V             | 96.990 | Hot Spot |\n",
       "| 10 | FT007615-05 | 5 | FT007615 | 100 | 200904 | MET_Y1230S             | 96.985 | Hot Spot |\n",
       "\n"
      ],
      "text/plain": [
       "   BATCH_ID    BATCH_NUM COMPD_ID CONC EXPERIMENT_ID KINASE                \n",
       "1  FT007615-05 5         FT007615 100  200904        MET                   \n",
       "2  FT007615-05 5         FT007615 100  200904        MET_L1195F            \n",
       "3  FT007615-05 5         FT007615 100  200904        MET_D1228Y            \n",
       "4  FT007615-05 5         FT007615 100  200904        MET_M1250I            \n",
       "5  FT007615-05 5         FT007615 100  200904        MET-TFG               \n",
       "6  FT007615-05 5         FT007615 100  200904        MET_D1228H            \n",
       "7  FT007615-05 5         FT007615 100  200904        MET_D1228N            \n",
       "8  FT007615-05 5         FT007615 100  200904        MET-KIF5B (Kex24Mex14)\n",
       "9  FT007615-05 5         FT007615 100  200904        MET_D1228V            \n",
       "10 FT007615-05 5         FT007615 100  200904        MET_Y1230S            \n",
       "   PCT_INHIBITION_AVG TECHNOLOGY\n",
       "1  99.330             Hot Spot  \n",
       "2  99.020             Hot Spot  \n",
       "3  98.955             Hot Spot  \n",
       "4  98.760             Hot Spot  \n",
       "5  98.510             Hot Spot  \n",
       "6  98.395             Hot Spot  \n",
       "7  97.520             Hot Spot  \n",
       "8  97.020             Hot Spot  \n",
       "9  96.990             Hot Spot  \n",
       "10 96.985             Hot Spot  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(kinasedata, 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "4dc622de-b61e-4a65-869d-73069b19282e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>HGNC_ID</th><th scope=col>SYMBOL</th><th scope=col>NAME</th><th scope=col>ALIAS_SYMBOL</th><th scope=col>ALIAS_NAME</th><th scope=col>PREV_SYMBOL</th><th scope=col>PREV_NAME</th><th scope=col>GENE_FAMILY</th><th scope=col>SYMBOL_UPPER</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>HGNC:19   </td><td>AANAT </td><td>aralkylamine N-acetyltransferase                   </td><td>SNAT               </td><td>serotonin N-acetyltransferase                      </td><td>NA     </td><td>arylalkylamine N-acetyltransferase                                             </td><td>GCN5 related N-acetyltransferases   </td><td>AANAT </td></tr>\n",
       "\t<tr><td>HGNC:15886</td><td>AAR2  </td><td>AAR2 splicing factor homolog                       </td><td>bA234K24.2         </td><td>NA                                                 </td><td>C20orf4</td><td>chromosome 20 open reading frame 4|AAR2 splicing factor homolog (S. cerevisiae)</td><td>NA                                  </td><td>AAR2  </td></tr>\n",
       "\t<tr><td>HGNC:33842</td><td>AARD  </td><td>alanine and arginine rich domain containing protein</td><td>LOC441376          </td><td>Alanine and arginine-rich domain-containing protein</td><td>C8orf85</td><td>chromosome 8 open reading frame 85                                             </td><td>NA                                  </td><td>AARD  </td></tr>\n",
       "\t<tr><td>HGNC:20   </td><td>AARS  </td><td>alanyl-tRNA synthetase                             </td><td>CMT2N|AlaRS        </td><td>alanine tRNA ligase 1, cytoplasmic                 </td><td>NA     </td><td>NA                                                                             </td><td>Aminoacyl tRNA synthetases, Class II</td><td>AARS  </td></tr>\n",
       "\t<tr><td>HGNC:21022</td><td>AARS2 </td><td>alanyl-tRNA synthetase 2, mitochondrial            </td><td>KIAA1270|bA444E17.1</td><td>alanine tRNA ligase 2, mitochondrial               </td><td>AARSL  </td><td>alanyl-tRNA synthetase like|alanyl-tRNA synthetase 2, mitochondrial (putative) </td><td>Aminoacyl tRNA synthetases, Class II</td><td>AARS2 </td></tr>\n",
       "\t<tr><td>HGNC:28417</td><td>AARSD1</td><td>alanyl-tRNA synthetase domain containing 1         </td><td>MGC2744            </td><td>NA                                                 </td><td>NA     </td><td>NA                                                                             </td><td>NA                                  </td><td>AARSD1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " HGNC\\_ID & SYMBOL & NAME & ALIAS\\_SYMBOL & ALIAS\\_NAME & PREV\\_SYMBOL & PREV\\_NAME & GENE\\_FAMILY & SYMBOL\\_UPPER\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t HGNC:19    & AANAT  & aralkylamine N-acetyltransferase                    & SNAT                & serotonin N-acetyltransferase                       & NA      & arylalkylamine N-acetyltransferase                                              & GCN5 related N-acetyltransferases    & AANAT \\\\\n",
       "\t HGNC:15886 & AAR2   & AAR2 splicing factor homolog                        & bA234K24.2          & NA                                                  & C20orf4 & chromosome 20 open reading frame 4\\textbar{}AAR2 splicing factor homolog (S. cerevisiae) & NA                                   & AAR2  \\\\\n",
       "\t HGNC:33842 & AARD   & alanine and arginine rich domain containing protein & LOC441376           & Alanine and arginine-rich domain-containing protein & C8orf85 & chromosome 8 open reading frame 85                                              & NA                                   & AARD  \\\\\n",
       "\t HGNC:20    & AARS   & alanyl-tRNA synthetase                              & CMT2N\\textbar{}AlaRS         & alanine tRNA ligase 1, cytoplasmic                  & NA      & NA                                                                              & Aminoacyl tRNA synthetases, Class II & AARS  \\\\\n",
       "\t HGNC:21022 & AARS2  & alanyl-tRNA synthetase 2, mitochondrial             & KIAA1270\\textbar{}bA444E17.1 & alanine tRNA ligase 2, mitochondrial                & AARSL   & alanyl-tRNA synthetase like\\textbar{}alanyl-tRNA synthetase 2, mitochondrial (putative)  & Aminoacyl tRNA synthetases, Class II & AARS2 \\\\\n",
       "\t HGNC:28417 & AARSD1 & alanyl-tRNA synthetase domain containing 1          & MGC2744             & NA                                                  & NA      & NA                                                                              & NA                                   & AARSD1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 9\n",
       "\n",
       "| HGNC_ID &lt;chr&gt; | SYMBOL &lt;chr&gt; | NAME &lt;chr&gt; | ALIAS_SYMBOL &lt;chr&gt; | ALIAS_NAME &lt;chr&gt; | PREV_SYMBOL &lt;chr&gt; | PREV_NAME &lt;chr&gt; | GENE_FAMILY &lt;chr&gt; | SYMBOL_UPPER &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| HGNC:19    | AANAT  | aralkylamine N-acetyltransferase                    | SNAT                | serotonin N-acetyltransferase                       | NA      | arylalkylamine N-acetyltransferase                                              | GCN5 related N-acetyltransferases    | AANAT  |\n",
       "| HGNC:15886 | AAR2   | AAR2 splicing factor homolog                        | bA234K24.2          | NA                                                  | C20orf4 | chromosome 20 open reading frame 4|AAR2 splicing factor homolog (S. cerevisiae) | NA                                   | AAR2   |\n",
       "| HGNC:33842 | AARD   | alanine and arginine rich domain containing protein | LOC441376           | Alanine and arginine-rich domain-containing protein | C8orf85 | chromosome 8 open reading frame 85                                              | NA                                   | AARD   |\n",
       "| HGNC:20    | AARS   | alanyl-tRNA synthetase                              | CMT2N|AlaRS         | alanine tRNA ligase 1, cytoplasmic                  | NA      | NA                                                                              | Aminoacyl tRNA synthetases, Class II | AARS   |\n",
       "| HGNC:21022 | AARS2  | alanyl-tRNA synthetase 2, mitochondrial             | KIAA1270|bA444E17.1 | alanine tRNA ligase 2, mitochondrial                | AARSL   | alanyl-tRNA synthetase like|alanyl-tRNA synthetase 2, mitochondrial (putative)  | Aminoacyl tRNA synthetases, Class II | AARS2  |\n",
       "| HGNC:28417 | AARSD1 | alanyl-tRNA synthetase domain containing 1          | MGC2744             | NA                                                  | NA      | NA                                                                              | NA                                   | AARSD1 |\n",
       "\n"
      ],
      "text/plain": [
       "  HGNC_ID    SYMBOL NAME                                               \n",
       "1 HGNC:19    AANAT  aralkylamine N-acetyltransferase                   \n",
       "2 HGNC:15886 AAR2   AAR2 splicing factor homolog                       \n",
       "3 HGNC:33842 AARD   alanine and arginine rich domain containing protein\n",
       "4 HGNC:20    AARS   alanyl-tRNA synthetase                             \n",
       "5 HGNC:21022 AARS2  alanyl-tRNA synthetase 2, mitochondrial            \n",
       "6 HGNC:28417 AARSD1 alanyl-tRNA synthetase domain containing 1         \n",
       "  ALIAS_SYMBOL        ALIAS_NAME                                         \n",
       "1 SNAT                serotonin N-acetyltransferase                      \n",
       "2 bA234K24.2          NA                                                 \n",
       "3 LOC441376           Alanine and arginine-rich domain-containing protein\n",
       "4 CMT2N|AlaRS         alanine tRNA ligase 1, cytoplasmic                 \n",
       "5 KIAA1270|bA444E17.1 alanine tRNA ligase 2, mitochondrial               \n",
       "6 MGC2744             NA                                                 \n",
       "  PREV_SYMBOL\n",
       "1 NA         \n",
       "2 C20orf4    \n",
       "3 C8orf85    \n",
       "4 NA         \n",
       "5 AARSL      \n",
       "6 NA         \n",
       "  PREV_NAME                                                                      \n",
       "1 arylalkylamine N-acetyltransferase                                             \n",
       "2 chromosome 20 open reading frame 4|AAR2 splicing factor homolog (S. cerevisiae)\n",
       "3 chromosome 8 open reading frame 85                                             \n",
       "4 NA                                                                             \n",
       "5 alanyl-tRNA synthetase like|alanyl-tRNA synthetase 2, mitochondrial (putative) \n",
       "6 NA                                                                             \n",
       "  GENE_FAMILY                          SYMBOL_UPPER\n",
       "1 GCN5 related N-acetyltransferases    AANAT       \n",
       "2 NA                                   AAR2        \n",
       "3 NA                                   AARD        \n",
       "4 Aminoacyl tRNA synthetases, Class II AARS        \n",
       "5 Aminoacyl tRNA synthetases, Class II AARS2       \n",
       "6 NA                                   AARSD1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "HGNC.t <- HGNC %>%  mutate(SYMBOL_UPPER = stringr::str_to_upper(SYMBOL))\n",
    "head(HGNC.t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "3cf1cdbb-b34b-4f05-91c1-ccea1e9258b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SYMBOL_UPPER</th><th scope=col>HGNC_ID</th><th scope=col>SYMBOL</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>AANAT </td><td>HGNC:19   </td><td>AANAT </td></tr>\n",
       "\t<tr><td>AAR2  </td><td>HGNC:15886</td><td>AAR2  </td></tr>\n",
       "\t<tr><td>AARD  </td><td>HGNC:33842</td><td>AARD  </td></tr>\n",
       "\t<tr><td>AARS  </td><td>HGNC:20   </td><td>AARS  </td></tr>\n",
       "\t<tr><td>AARS2 </td><td>HGNC:21022</td><td>AARS2 </td></tr>\n",
       "\t<tr><td>AARSD1</td><td>HGNC:28417</td><td>AARSD1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " SYMBOL\\_UPPER & HGNC\\_ID & SYMBOL\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t AANAT  & HGNC:19    & AANAT \\\\\n",
       "\t AAR2   & HGNC:15886 & AAR2  \\\\\n",
       "\t AARD   & HGNC:33842 & AARD  \\\\\n",
       "\t AARS   & HGNC:20    & AARS  \\\\\n",
       "\t AARS2  & HGNC:21022 & AARS2 \\\\\n",
       "\t AARSD1 & HGNC:28417 & AARSD1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 3\n",
       "\n",
       "| SYMBOL_UPPER &lt;chr&gt; | HGNC_ID &lt;chr&gt; | SYMBOL &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| AANAT  | HGNC:19    | AANAT  |\n",
       "| AAR2   | HGNC:15886 | AAR2   |\n",
       "| AARD   | HGNC:33842 | AARD   |\n",
       "| AARS   | HGNC:20    | AARS   |\n",
       "| AARS2  | HGNC:21022 | AARS2  |\n",
       "| AARSD1 | HGNC:28417 | AARSD1 |\n",
       "\n"
      ],
      "text/plain": [
       "  SYMBOL_UPPER HGNC_ID    SYMBOL\n",
       "1 AANAT        HGNC:19    AANAT \n",
       "2 AAR2         HGNC:15886 AAR2  \n",
       "3 AARD         HGNC:33842 AARD  \n",
       "4 AARS         HGNC:20    AARS  \n",
       "5 AARS2        HGNC:21022 AARS2 \n",
       "6 AARSD1       HGNC:28417 AARSD1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ht <- HGNC.t %>%\n",
    "    select(SYMBOL_UPPER, HGNC_ID, SYMBOL)\n",
    "head(ht)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "4ea179c6-5a87-4ae5-9f7d-d8bd7bdabc58",
   "metadata": {},
   "outputs": [],
   "source": [
    "# regex string generator to find partial matches\n",
    "get_match_indices <- function(ls_sp1, ls_sp2, match_values, i) {\n",
    "    lower_sp1 <- tolower(substr(ls_sp1[i], 2, nchar(ls_sp1[i])))\n",
    "    first_sp1 <- substr(ls_sp1[i], 1, 1)\n",
    "    sp2 <- ls_sp2[i]\n",
    "    pattern = paste0(\"(.*[\",\n",
    "                       toupper(first_sp1), \n",
    "                       \"|\", \n",
    "                       tolower(first_sp1),\n",
    "                       \"]\", \n",
    "                       lower_sp1,\n",
    "                       \".*\", sp2, \")\"\n",
    "                     )\n",
    "    print(pattern)\n",
    "    match_indices <- grep(pattern = pattern,\n",
    "                          ignore.case = F, \n",
    "                          x = match_values \n",
    "                         )\n",
    "    print(paste(i, \":\", match_indices))\n",
    "    return(match_indices)\n",
    "    }\n",
    "\n",
    "# main call to loop thru indices for a 1 index match\n",
    "map_regex_match <- function(df, HGNC, i) {\n",
    "    ls_sp1 <- df$sp1\n",
    "    ls_sp2 <- df$sp2\n",
    "    name_match_indices <- get_match_indices(ls_sp1, ls_sp2, HGNC$NAME, i)\n",
    "    if (length(name_match_indices) == 1) {\n",
    "        return(name_match_indices)\n",
    "    } else {\n",
    "        if (length(name_match_indices > 1)) {\n",
    "            for (j in name_match_indices) {\n",
    "                print(j)\n",
    "                print(HGNC$NAME[j])\n",
    "                alias_match_indices <- get_match_indices(ls_sp1, ls_sp2, HGNC$ALIAS_NAME[j], i)\n",
    "                if (length(alias_match_indices) == 1) {\n",
    "                    return(j)\n",
    "                } else {\n",
    "                    prev_match_indices <- get_match_indices(ls_sp1, ls_sp2, HGNC$PREV_NAME[j], i)\n",
    "                    if (length(prev_match_indices) == 1) {\n",
    "                        return(j)\n",
    "                    }\n",
    "                }\n",
    "            }\n",
    "        }\n",
    "    }\n",
    "    return(NA)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "bbf7a3bc-9b00-4ab4-b396-181892e1c373",
   "metadata": {},
   "outputs": [],
   "source": [
    "regex_alias <- function(the_string) {\n",
    "   # rtn_ls <- list(SYMBOL = NA, HGNC_ID = NA)\n",
    "   rtn_c <- c(\"NA\", \"NA\") \n",
    "   if (the_string == \"NA\") {return(rtn_c)}\n",
    "   pattern <- paste0(\"(?<![A-Za-z])\", the_string, \"(?![A-Za-z])\") \n",
    "   colnames <- c(\"SYMBOL\", \"ALIAS_SYMBOL\", \"PREV_SYMBOL\", \"NAME\", \"PREV_NAME\", \"ALIAS_NAME\") \n",
    "   nbr_rows <- -1\n",
    "   cnt <- 1\n",
    "   nbr_of_fields <- length(colnames)\n",
    "    \n",
    "   while (nbr_rows != 1) {\n",
    "       hgnc_df <- HGNC %>%\n",
    "                filter(grepl(x = !!as.name(colnames[cnt]), \n",
    "                            pattern = pattern,\n",
    "                            perl = T,\n",
    "                            ignore.case = T)\n",
    "                       )\n",
    "                    \n",
    "       # print(hgnc_df)\n",
    "        nbr_rows <- nrow(hgnc_df)\n",
    "        if (cnt == nbr_of_fields) {break}\n",
    "        cnt <- cnt + 1\n",
    "   }\n",
    "    \n",
    "    \n",
    "   if (nrow(hgnc_df) != 0) {\n",
    "       rtn_c <- c(hgnc_df$SYMBOL, hgnc_df$HGNC_ID)\n",
    "   }\n",
    "   # rtn_ls$SYMBOL <- hgnc_df$SYMBOL\n",
    "   # rtn_ls$HGNC_ID <- hgnc_df$HGNC_ID\n",
    "    \n",
    "   # return(rtn_ls)\n",
    "   return(rtn_c)\n",
    "                                 \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "a078489e-4253-408a-b905-279df7f4fbe2",
   "metadata": {},
   "outputs": [],
   "source": [
    "get.hgnc.id <- function(HGNC, pattern ) {\n",
    "    d <- HGNC %>% filter(grepl(x = SYMBOL, pattern = paste0(\"^\", pattern, \"$\"))) %>% select(HGNC_ID) %>% pull() \n",
    "    # print(pattern)\n",
    "    # print(d)\n",
    "    if (length(d) > 1) {\n",
    "        return(\"\")\n",
    "        }\n",
    "    if (length(d) == 0) {\n",
    "        return(\"\")\n",
    "        }\n",
    "    return(d)\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "68ae398f-6944-4545-b1ba-4b9b0bbbaf94",
   "metadata": {},
   "outputs": [],
   "source": [
    "ht <- HGNC %>%\n",
    "    select(HGNC_ID, SYMBOL) %>%\n",
    "    mutate(SYMBOL_UPPER = stringr::str_to_upper(SYMBOL))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "01c9d620-31c0-45db-a3c7-eefa60c44790",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dd <- read.csv(\"/home/jovyan/work/Downloads/output-df.csv\")\n",
    "# dt <- read.csv(\"/home/jovyan/work/Downloads/output.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "81d889d8-802b-4ea2-b9d3-fbd892332cde",
   "metadata": {},
   "outputs": [],
   "source": [
    "dff <-\n",
    "kinasedata %>% \n",
    "    select(KINASE, PCT_INHIBITION_AVG) %>%\n",
    "    filter(PCT_INHIBITION_AVG >= 60,\n",
    "        !grepl(pattern = \"\\\\[\", x = KINASE), \n",
    "        !grepl(pattern = \"^(?!.*\\\\/).*\\\\(.*\\\\)$\", x = KINASE, perl=T) ) %>%\n",
    "    mutate(No_Greek =  case_when(\n",
    "        grepl(pattern=\"α\", KINASE) ~ gsub(\"α\", \"A\", KINASE),\n",
    "        grepl(pattern = \"β\", KINASE)  ~ gsub(\"β\", \"B\", KINASE),\n",
    "        grepl(pattern = \"γ\", KINASE) ~ gsub(\"γ\", \"G\", KINASE),\n",
    "        grepl(pattern = \"δ\", KINASE) ~ gsub(\"δ\", \"d\", KINASE),\n",
    "        grepl(pattern = \"ᐃ\" , KINASE)~ gsub(\"ᐃ\", \"D\",KINASE),\n",
    "        grepl(pattern = \"ε\" , KINASE)~ gsub(\"ε\", \"E\",KINASE),\n",
    "        grepl(pattern = \"ζ\" , KINASE)~ gsub(\"ζ\", \"Z\",KINASE),\n",
    "        grepl(pattern = \"η\" , KINASE)~ gsub(\"η\",\"H\", KINASE),\n",
    "        grepl(pattern = \"θ\" , KINASE)~ gsub(\"θ\", \"Q\",KINASE),\n",
    "        grepl(pattern = \"ι\" , KINASE)~ gsub(\"ι\", \"I\",KINASE),\n",
    "        TRUE ~ KINASE ) ) %>%\n",
    "    mutate(No_Space = str_replace(No_Greek, \" \", \"_\"),\n",
    "    Just_KINASE = sub(\"_.*\", \"\", No_Space),\n",
    "    Just_KINASE = stringr::str_to_upper(Just_KINASE)) %>%\n",
    "    separate(No_Space, into = c('sp1', 'sp2', 'sp3'), extra = 'drop', remove = FALSE, fill=\"right\") %>%\n",
    "    merge(ht, by.x = 'Just_KINASE', by.y = 'SYMBOL_UPPER', all.x = TRUE, all.y = FALSE) %>%\n",
    "    arrange(desc(PCT_INHIBITION_AVG)) %>%\n",
    "    rename(Result = PCT_INHIBITION_AVG, Kinase = KINASE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "3dfd8bba-8fbc-4e86-aad4-0062ab81d68e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "dtt <- dff %>% \n",
    "                filter(!is.na(sp2) | !is.na(sp3), \n",
    "                             is.na(HGNC_ID)) %>%\n",
    "                mutate(INDEX = row_number())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "306f27bd-ac34-4006-bf75-00d540cf789a",
   "metadata": {},
   "outputs": [],
   "source": [
    "dtt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6dd2e852-ba40-4ccc-9834-957b422586b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "regex_iterate_match(p2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8726ed3-21ae-430e-a3b8-09eeb4669b08",
   "metadata": {},
   "outputs": [],
   "source": [
    "get.hgnc.id(HGNC, 'AMPKA2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a0c1113-d159-432b-94f2-842c8da21de0",
   "metadata": {},
   "outputs": [],
   "source": [
    "for (i in seq(nrow(dtt))) {\n",
    "        p1 = dtt %>% filter(INDEX == i) %>% select(sp1) %>% pull()\n",
    "        x1 = get.hgnc.id(HGNC, p1)\n",
    "        print(paste0('x1 top: ', x1))\n",
    "        if (x1 == \"\") {\n",
    "                p2 = dtt %>% filter(INDEX == i) %>% select(sp2) %>% pull()\n",
    "                x2 = get.hgnc.id(HGNC, p2)\n",
    "                print(paste0('x2 top: ', x2))\n",
    "                if (x2 == \"\") {\n",
    "                    x1 <- regex_iterate_match(p1) \n",
    "                    print(paste0('x1 bottom:', x1))\n",
    "                    if (is.na(x1)) {\n",
    "                        x1 = regex_iterate_match(p2)\n",
    "                        print(paste0('x1:', x1, ' p2: ', p2))\n",
    "                    }\n",
    "                    }\n",
    "                else { x1 = duplicate(x2, shallow=F) }\n",
    "                }\n",
    "        if (i > 11) {\n",
    "        print(paste0(\"x1:\", x1, \"i: \", i))\n",
    "        if (!is.na(x1)) {\n",
    "                tmp.symbol <- HGNC %>% filter(HGNC_ID == x1) %>% select(SYMBOL) %>% pull()\n",
    "                # print(paste0(\"tmp:\", x1))\n",
    "                dd <- dd %>% mutate(HGNC_ID = replace(HGNC_ID, Result == dtt$Result[i], x1),\n",
    "                               SYMBOL = replace(SYMBOL, Result == dtt$Result[i], tmp.symbol))\n",
    "                }\n",
    "        }\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8da5aa2b-5d5a-40d9-81d4-ebc7c3a54d47",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for (i in seq(nrow(dt))) {\n",
    "        p1 = dt %>% filter(INDEX == i) %>% select(sp1) %>% pull()\n",
    "        x1 = get.hgnc.id(HGNC, p1)\n",
    "        print(paste0('x1 top: ', x1))\n",
    "        if (x1 == \"\") {\n",
    "                p2 = dt %>% filter(INDEX == i) %>% select(sp2) %>% pull()\n",
    "                x2 = get.hgnc.id(HGNC, p2)\n",
    "                print(paste0('x2 top: ', x2))\n",
    "                if (x2 == \"\") {\n",
    "                    x1 <- regex_iterate_match(p1) \n",
    "                    print(paste0('x1 bottom:', x1))\n",
    "                    if (is.na(x1)) {\n",
    "                        x1 = regex_iterate_match(p2)\n",
    "                        print(paste0('x1:', x1, ' p2: ', p2))\n",
    "                    }\n",
    "                    }\n",
    "                else { x1 = duplicate(x2, shallow=F) }\n",
    "                }\n",
    "        if (i > 11) {\n",
    "        print(paste0(\"x1:\", x1, \"i: \", i))\n",
    "        if (!is.na(x1)) {\n",
    "                tmp.symbol <- HGNC %>% filter(HGNC_ID == x1) %>% select(SYMBOL) %>% pull()\n",
    "                # print(paste0(\"tmp:\", x1))\n",
    "                dd <- dd %>% mutate(HGNC_ID = replace(HGNC_ID, Result == dt$Result[i], x1),\n",
    "                               SYMBOL = replace(SYMBOL, Result == dt$Result[i], tmp.symbol))\n",
    "                }\n",
    "        }\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "4e2c2d57-6948-4ec2-ab14-7a42042f5730",
   "metadata": {},
   "outputs": [],
   "source": [
    "regex_iterate_match <- function(pattern) {\n",
    "    colnames <- c(\"SYMBOL\", \"ALIAS_SYMBOL\", \"PREV_SYMBOL\", \"NAME\", \"PREV_NAME\", \"ALIAS_NAME\") \n",
    "    rtn <- NA\n",
    "    nbr_rows <- -1\n",
    "    cnt <- 1\n",
    "    nbr_of_fields <- length(colnames)\n",
    "    pattern = paste0(\"(^|\\\\|)\", pattern, \"(\\\\||$)\")\n",
    "\n",
    "    while (nbr_rows != 1) {\n",
    "       hgnc_df <- HGNC %>%\n",
    "                filter(grepl(x = !!as.name(colnames[cnt]), \n",
    "                            pattern = pattern,\n",
    "                            perl = T, \n",
    "                            ignore.case = T)\n",
    "                       )\n",
    "\n",
    "        # print(hgnc_df)\n",
    "        nbr_rows <- nrow(hgnc_df)\n",
    "        if (nbr_rows == 1) {\n",
    "            return( hgnc_df$HGNC_ID)\n",
    "            }\n",
    "        if (cnt == nbr_of_fields) {break}\n",
    "        cnt <- cnt + 1\n",
    "    }\n",
    "    return( rtn)\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11c3e3b1-c99d-439b-b2cd-85b85a76d360",
   "metadata": {},
   "outputs": [],
   "source": [
    "dd %>% filter(is.na(SYMBOL))#, !is.na(sp2), nchar(sp2) < 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "979f794a-839b-4191-8fc7-d14ec1788509",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sym_nomatch <- dff %>% \n",
    "        filter(is.na(HGNC_ID))\n",
    "\n",
    "df_sym_match <- dff %>% \n",
    "        filter(!is.na(HGNC_ID))  \n",
    "\n",
    "SEC_HGNC <- HGNC %>% separate_rows(ALIAS_SYMBOL, sep = \"\\\\|\") %>% \n",
    "                                        separate_rows(PREV_SYMBOL, sep = \"\\\\|\") %>%\n",
    "                                        mutate(ALIAS_UPPER = stringr::str_to_upper(ALIAS_SYMBOL), \n",
    "                                                     PREV_UPPER = stringr::str_to_upper(PREV_SYMBOL))\n",
    "\n",
    "df_sym_nomatch <- df_sym_nomatch %>%\n",
    "                                mutate(HGNC_ID = NULL,\n",
    "                                             SYMBOL = NULL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "da5a1062-f705-411c-a908-ae214a934723",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Just_KINASE</th><th scope=col>Kinase</th><th scope=col>Result</th><th scope=col>No_Greek</th><th scope=col>No_Space</th><th scope=col>sp1</th><th scope=col>sp2</th><th scope=col>sp3</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " Just\\_KINASE & Kinase & Result & No\\_Greek & No\\_Space & sp1 & sp2 & sp3\\\\\n",
       " <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 8\n",
       "\n",
       "| Just_KINASE &lt;chr&gt; | Kinase &lt;chr&gt; | Result &lt;dbl&gt; | No_Greek &lt;chr&gt; | No_Space &lt;chr&gt; | sp1 &lt;chr&gt; | sp2 &lt;chr&gt; | sp3 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     Just_KINASE Kinase Result No_Greek No_Space sp1 sp2 sp3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sym_nomatch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "666fe203-8231-4eee-8703-d4dbc72ddec8",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_alias_match <- merge(df_sym_nomatch, \n",
    "                        SEC_HGNC, \n",
    "                        by.x = 'Just_KINASE', \n",
    "                        by.y = 'ALIAS_UPPER', \n",
    "                        all.x = T, \n",
    "                        all.y = F\n",
    "                        )\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
